docs/docs/files-gatsby-looks-for-in-a-plugin.md

```diff
@@ -20,3 +33,32 @@ All files are optional unless specifically marked as required.
 - `gatsby-browser.js` — usage details are in the [browser API reference](/docs/browser-apis/)
 - `gatsby-node.js` — usage details are in the [Node API reference](/docs/node-apis/)
 - `gatsby-ssr.js` — usage details are in the [SSR API reference](/docs/ssr-apis/)
+
+In a generic plugin the `gatsby-node.js` file enables the use of [gatsby node APIs](/docs/node-apis/) which is where all the processes like createPage,createResolvers,sourceNodes exist.
+
+In this file you are able to carry out functions that load API keys, send calls to APIs,create Gatsby-nodes using the API response, Create pages to display responses and loop through responses to display an individual response. (A good use case of this would be a plugin that gets data from an API.)
```

Comment:
```suggestion
In this file you are able to carry out functions that load API keys, send calls to APIs,  create Gatsby-nodes using the API response, create pages to display responses, and loop through responses to display an individual response (a good use case of this would be a plugin that gets data from an API).
```